gp-templates/helper-functions.php

```diff
@@ -172,44 +172,91 @@ function textareas( $entry, $permissions, $index = 0 ) {
 	$disabled = $can_edit ? '' : 'disabled="disabled"';
 	?>
 	<div class="textareas">
-		<?php
+	<?php
 		if ( isset( $entry->warnings[ $index ] ) ) :
 			$referenceable = $entry->warnings[ $index ];
 
 			foreach ( $referenceable as $key => $value ) :
-			?>
+	?>
 				<div class="warning secondary">
 					<strong><?php _e( 'Warning:', 'glotpress' ); ?></strong> <?php echo esc_html( $value ); ?>
 
 					<?php if ( $can_approve ) : ?>
 						<a href="#" class="discard-warning" data-nonce="<?php echo esc_attr( wp_create_nonce( 'discard-warning_' . $index . $key ) ); ?>" data-key="<?php echo esc_attr( $key ); ?>" data-index="<?php echo esc_attr( $index ); ?>"><?php _e( 'Discard', 'glotpress' ); ?></a>
 					<?php endif; ?>
 				</div>
-		<?php
+	<?php
 			endforeach;
 
 		endif;
-		?>
+	?>
 		<blockquote class="translation"><em><small><?php echo prepare_original( esc_translation( gp_array_get( $entry->translations, $index ) ) ); // WPCS: XSS ok. ?></small></em></blockquote>
 		<textarea class="foreign-text" name="translation[<?php echo esc_attr( $entry->original_id ); ?>][]" id="translation_<?php echo esc_attr( $entry->original_id ); ?>_<?php echo esc_attr( $index ); ?>" <?php echo $disabled; // WPCS: XSS ok. ?>><?php echo gp_prepare_translation_textarea( esc_translation( gp_array_get( $entry->translations, $index ) ) ); // WPCS: XSS ok. ?></textarea>
 
 		<p>
-			<?php
-			if ( $can_edit ) {
-				gp_entry_actions();
-			}
-			elseif ( is_user_logged_in() ) {
-				_e( 'You are not allowed to edit this translation.', 'glotpress' );
-			}
-			else {
-				printf( __( 'You <a href="%s">have to log in</a> to edit this translation.', 'glotpress' ), esc_url( wp_login_url( gp_url_current() ) ) );
-			}
-			?>
+	<?php
+	if ( $can_edit ) {
+		gp_entry_actions();
+	}
+	elseif ( is_user_logged_in() ) {
+		_e( 'You are not allowed to edit this translation.', 'glotpress' );
+	}
+	else {
+		printf( __( 'You <a href="%s">have to log in</a> to edit this translation.', 'glotpress' ), esc_url( wp_login_url( gp_url_current() ) ) );
+	}
+	?>
 		</p>
 	</div>
 	<?php
 }
 
+function feedbacks( $entry, $permissions ) {
+	list( $can_edit, $can_approve ) = $permissions;
+	$feedback = get_feedback( $entry );
+?>
+	<dl>
+	<?php
+		$disabled = '';
+		if ( ! empty( $feedback ) && $feedback[ 'user_id' ] !== get_current_user_id() ) {
+			$disabled = $can_edit ? '' : 'disabled="disabled"';
+		}
+		echo '<dt>' . __( 'Reviewers feedback:', 'glotpress' ) . '</dt>';
+
+		if ( ! empty( $feedback ) ) {
+			$feedback = $feedback[ 'feedback' ];
+		}
+
+		$button_label = __( 'Add feedback', 'glotpress' );
+	?>
+	<?php
+		if ( GP::$permission->current_user_can( 'admin', 'feedbacks', $project_id ) ) {
```

Comment:
`$project_id` isn't defined here you have to look up the project first.

Reply:
fixed